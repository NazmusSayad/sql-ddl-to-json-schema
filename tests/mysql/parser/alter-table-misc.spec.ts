import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Parser: Should alter table algorithm, with online, ignore and wait keywords.': {
    queries: [
      `ALTER online ignore TABLE people wait 5 algorithm = default;`,
      `ALTER TABLE people algorithm=default;`,
      `ALTER TABLE people algorithm default;`,
    ],
  },

  'Parser: Should alter column default value.': {
    queries: [
      `ALTER TABLE people alter column age set default 18;`,
      `ALTER TABLE people alter age set default 18;`,
    ],
  },

  'Parser: Should drop column default value.': {
    queries: [
      `ALTER TABLE people alter column name drop default;`,
      `ALTER TABLE people alter name drop default;`,
    ],
  },

  'Parser: Should alter table default character set and collation.': {
    queries: [
      `ALTER TABLE people default character set utf8 collate utf8_general_ci;`,
      `ALTER TABLE people character set = utf8 collate = utf8_general_ci;`,
      `ALTER TABLE people character set=utf8 collate=utf8_general_ci;`,
    ],
  },

  'Parser: Should alter table default character set.': {
    queries: [
      `ALTER TABLE people default character set utf8;`,
      `ALTER TABLE people character set = utf8;`,
      `ALTER TABLE people character set=utf8;`,
    ],
  },

  'Parser: Should alter table converting to character set and collation.': {
    queries: [`ALTER TABLE people convert to character set utf8 collate utf8_general_ci;`],
  },

  'Parser: Should alter table converting to character set.': {
    queries: [`ALTER TABLE people convert to character set utf8;`],
  },

  'Parser: Should run several simple alter table statements.': {
    queries: [
      `
      ALTER TABLE people enable keys;
      ALTER TABLE people disable keys;
      ALTER TABLE people discard tablespace;
      ALTER TABLE people import tablespace;
      ALTER TABLE people force;
      ALTER TABLE people with validation;
      ALTER TABLE people without validation;
      ALTER TABLE people rename to persons;
      ALTER TABLE people rename persons;
      `,
    ],
  },

  'Parser: Should run one alter table statement with several alter table specs and options.': {
    queries: [
      `
      ALTER TABLE people comment 'test', enable keys , disable keys,discard tablespace,
      import tablespace ,force, comment 'another test',
      with validation, without validation, rename to persons
      ,rename persons;
      `,
    ],
  },

  'Parser: Should run one alter table statement with only one option.': {
    queries: [
      `
      ALTER TABLE people comment 'test';
      `,
    ],
  },

  'Parser: Should alter table lock.': {
    queries: [
      `ALTER TABLE people lock default;`,
      `ALTER TABLE people lock=default;`,
      `ALTER TABLE people lock = default;`,
    ],
  },

  'Parser: Should alter table ordering by several columns.': {
    queries: [`ALTER TABLE people order by id, initials,name ,fullname , family;`],
  },

  'Parser: Should alter table ordering by one column.': {
    queries: [`ALTER TABLE people order by id;`],
  },

  'Parser: Should alter table adding period for system_time.': {
    queries: [`alter table test add period for system_time (test1, test2);`],
  },
});
