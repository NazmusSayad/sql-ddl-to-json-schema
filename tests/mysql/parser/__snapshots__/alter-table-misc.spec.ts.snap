// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser: Should alter column default value. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "setDefaultColumnValue",
                    "column": "age",
                    "value": 18
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter column default value. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "setDefaultColumnValue",
                    "column": "age",
                    "value": 18
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table adding period for system_time. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "test",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "addPeriodForSystemTime",
                    "startColumnName": "test1",
                    "endColumnName": "test2"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table algorithm, with online, ignore and wait keywords. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeAlgorithm",
                    "algorithm": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table algorithm, with online, ignore and wait keywords. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeAlgorithm",
                    "algorithm": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table algorithm, with online, ignore and wait keywords. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeAlgorithm",
                    "algorithm": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table converting to character set and collation. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "convertToCharacterSet",
                    "charset": "utf8",
                    "collate": "utf8_general_ci"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table converting to character set. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "convertToCharacterSet",
                    "charset": "utf8",
                    "collate": null
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set and collation. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    },
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "collation": "utf8_general_ci"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set and collation. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    },
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "collation": "utf8_general_ci"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set and collation. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    },
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "collation": "utf8_general_ci"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table default character set. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "charset": "utf8"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table lock. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeLock",
                    "lock": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table lock. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeLock",
                    "lock": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table lock. 3`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "changeLock",
                    "lock": "default"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table ordering by one column. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "orderBy",
                    "columns": [
                      "id"
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should alter table ordering by several columns. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "orderBy",
                    "columns": [
                      "id",
                      "initials",
                      "name",
                      "fullname",
                      "family"
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop column default value. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropDefaultColumnValue",
                    "column": "name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should drop column default value. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "dropDefaultColumnValue",
                    "column": "name"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should run one alter table statement with only one option. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "comment": "test"
                      }
                    }
                  ]
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should run one alter table statement with several alter table specs and options. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "comment": "test"
                      }
                    }
                  ]
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "enableKeys"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "disableKeys"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "discardTablespace"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "importTablespace"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "force"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "tableOptions": {
                  "id": "P_CREATE_TABLE_OPTIONS",
                  "def": [
                    {
                      "id": "O_CREATE_TABLE_OPTION",
                      "def": {
                        "comment": "another test"
                      }
                    }
                  ]
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "withValidation"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "withoutValidation"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "rename",
                    "newName": "persons"
                  }
                }
              }
            },
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "rename",
                    "newName": "persons"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;

exports[`Parser: Should run several simple alter table statements. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "enableKeys"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "disableKeys"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "discardTablespace"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "importTablespace"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "force"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "withValidation"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "withoutValidation"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "rename",
                    "newName": "persons"
                  }
                }
              }
            }
          ]
        }
      }
    },
    {
      "id": "P_DDS",
      "def": {
        "id": "P_ALTER_TABLE",
        "def": {
          "table": "people",
          "specs": [
            {
              "id": "P_ALTER_TABLE_SPECS",
              "def": {
                "spec": {
                  "id": "O_ALTER_TABLE_SPEC",
                  "def": {
                    "action": "rename",
                    "newName": "persons"
                  }
                }
              }
            }
          ]
        }
      }
    }
  ]
}
`;
